sentry-android-core/src/test/java/io/sentry/android/core/ConnectivityCheckerTest.kt

```diff
@@ -1,31 +1,161 @@
 import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
 import io.sentry.android.core.util.ConnectivityChecker
-import kotlin.test.BeforeTest
 import kotlin.test.Test
-import org.junit.runner.RunWith
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
 
-@RunWith(AndroidJUnit4::class)
 class ConnectivityCheckerTest {
 
-    private lateinit var context: Context
+    @Test
+    fun `When network is active and connected, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(true)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When network is active but not connected, return false for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(false)
+        assertFalse(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When there's no permission, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        whenever(contextMock.checkPermission(any(), any(), any())).thenReturn(PERMISSION_DENIED)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When network is not active, return false for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        assertFalse(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for isConnected`() {
+        assertNull(ConnectivityChecker.isConnected(mock(), mock()))
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
+    }
+
+    @Test
+    fun `When sdkInfoVersion is not min Marshmallow, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(0)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
```

Comment:
`l`: As already pointed out maybe it is better to use an enum for the return result.

Reply:
`event.contexts.device.connectionType` is a raw `String`, we can use an `enum`, but then we more complexibility with no big gain, it needs to be deserialized anyway.

Reply:
done